draft/webxdc-user-guide.md

```diff
@@ -0,0 +1,156 @@
+  that allow accessing resources only from the ZIP-file
+
+
+## Webxdc API
+
+Webxdc apps can do anything that is allowed in webviews -
+html, css, images, canvas, javascript and so on.
+
+There are some additional APIs available once `webxdc.js` is included
+(the file will be provided by the concrete implementations,
+no need to add `webxdc.js` to your ZIP-file):
+
+```html
+<script src="webxdc.js></script>
+```
+
+### sendUpdate()
+
+```js
+window.webxdc.sendUpdate(descr, payload);
+```
+
+Webxdc apps are usually shared in a chat and run independently on each peer.
+To get a shared state, the peers use `sendUpdate()` to send updates to each other.
+
+- `descr`: short, human-readable description what this update is about.
+  this is shown eg. as a fallback text in an email program.
+- `payload`: any javascript primitive, array or object.
+
+All peers, including the sending one,
+will receive the update by the callback given to `setUpdateListener()`.
+
+### setUpdateListener()
+
+```js
+window.webxdc.setUpdateListener((update) => {});
+```
+
+With `setUpdateListener()` you define a callback that receives the updates
+sent by `sendUpdate()`.
+
+- `update`: passed to the callback on updates.
+- `update.payload`: equals the payload given to `sendUpdate()`
+
+The callback is called for updates sent by you or other peers.
+
+
+### getAllUpdates()
+
+```
+payloads = window.webxdc.getAllUpdates()
+```
+
+In case your Webxdc was just started,
+you may want to reconstruct the state from the last run -
+and also incorporate updates that may have arrived while the app was not running.
+
+- `payloads`: the function returns all previous updates in an array, 
+  eg. `[{payload: "foo"},{payload: "bar"}]`
+  if `webxdc.sendUpdate("foo"); webxdc.sendUpdate("bar");` was called on the last run.
+
+
+### selfAddr()
+
+```js
+addr = window.webxdc.selfAddr()
+```
+
+Returns the peer's own address.
+This is esp. useful if you want to differ between different peers -
+just send the address along with the payload,
+and, if needed, compare the payload addresses against selfAddr() later on.
+
+
+### selfName()
+
+```js
+addr = window.webxdc.selfName()
+```
+
+Returns the peer's own name.
+This is name chosen by the user in their settings,
+if there is nothing set, that defaults to the peer's address.
+
+
+## manifest.toml
+
+If the ZIP-file contains a `manifest.toml` in its root directory,
+some basic information are read and used from there.
+
+the `manifest.toml` has the following format
+
+```toml
+name = "My App Name"
+icon = "icon.png"
+```
+
+- **name** - The name of the app.
+  If no name is set or if there is no manifest, the filename is used as the app name.
+- **icon** - The icon to use for the app.
+  The icon must be a `.png` or `.jpg` file and is read from the ZIP-file root directory.
+  The icon should be a square at reasonable width/height.
+  If no icon is set or if there is no manifest, a default icon will be used.
```

Comment:
as said above, maybe it would be better to leave this out and just go by file name here